### PR TITLE
Add regions config option

### LINKED ISSUE #989: Add ability to map region to policy without additional files
## Is your feature request related to a problem? Please describe.
I'm always frustrated when I think of `region_mapping` config. It shouldn't be an external file.

## Describe the solution you'd like
Make it possible to configure this via YAML.

### PR DESCRIPTION
**Pull request by smallhive**
Closes #989.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -261,6 +261,7 @@ placement_policy:
 |------------------|----------|---------------|---------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | `default`        | `string` | yes           | `REP 3`       | Default policy of placing containers in NeoFS. If a user sends a request `CreateBucket` and doesn't define policy for placing of a container in NeoFS, the S3 Gateway will put the container with default policy. |
 | `region_mapping` | `string` | yes           |               | Path to file that maps aws `LocationContraint` values to NeoFS placement policy. The similar to `--container-policy` flag in `neofs-s3-authmate` util, see in [docs](./authmate.md#containers-policy)             |
+| `regions`        | `map`    | yes           |               | The hashtable than maps aws `LocationContraint` values to NeoFS placement policy. The similar to `region_mapping` config option. The values extends/overrides values from region_mapping file, if it provided.    |
```
Comment (roman-khimov) [inline docs/configuration.md:264]: Hashtable that maps aws `LocationConstraint` values to NeoFS placement policy. It's similar to the `region_mapping` config option, but allows for in-place configuration and extends/overrides values from `region_mapping` file if it's provided.

File: docs/configuration.md [hunk 2]
```diff
@@ -261,6 +261,7 @@ placement_policy:
 |------------------|----------|---------------|---------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | `default`        | `string` | yes           | `REP 3`       | Default policy of placing containers in NeoFS. If a user sends a request `CreateBucket` and doesn't define policy for placing of a container in NeoFS, the S3 Gateway will put the container with default policy. |
 | `region_mapping` | `string` | yes           |               | Path to file that maps aws `LocationContraint` values to NeoFS placement policy. The similar to `--container-policy` flag in `neofs-s3-authmate` util, see in [docs](./authmate.md#containers-policy)             |
```
Comment (roman-khimov) [inline docs/configuration.md:263]: `Constraint` here as well (separate fix).

File: config/config.yaml [hunk 3]
```diff
@@ -127,6 +127,10 @@ placement_policy:
   # Region to placement policy mapping json file.
   # Path to container policy mapping. The same as '--container-policy' flag for authmate
   region_mapping: /path/to/container/policy.json
+  regions:
```
Comment (roman-khimov) [inline config/config.yaml:130]: Maybe call it `locations`, a bit more neutral? It's `LocationConstraint anyway`.

File: docs/configuration.md (hunk 1 above)
Comment (smallhive) [reply docs/configuration.md]: Updated

File: docs/configuration.md (hunk 2 above)
Comment (smallhive) [reply docs/configuration.md]: Extended

File: config/config.yaml (hunk 3 above)
Comment (smallhive) [reply config/config.yaml]: Renamed

File: docs/configuration.md (hunk 2 above)
Comment (roman-khimov) [reply docs/configuration.md]: Don't see this being fixed.

Comment (smallhive) [reply docs/configuration.md]: I've added extra info inside authmade.md, now I think I was a mistake.
Could you please extend the message? What the `Constraint`? In this place already mentioned `LocationContraint`

Comment (roman-khimov) [reply docs/configuration.md]: s/Contraint/Constraint/

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**